swift/apple/FirezoneNetworkExtension/Adapter.swift

```diff
@@ -205,9 +205,7 @@ class Adapter {
     // Cancel network monitor
     networkMonitor?.cancel()
 
-    // Shutdown the tunnel
-    session?.disconnect()
-
+    // Shutdown the tunnel, assigning nil to session will invoke `Drop` on the Rust side
     session = nil
```

Comment:
I think this might be redundant because Swift will call `deinit` recursively?

Reply:
Yeah I think this can be removed too - the only real risk is if the Adapter goes away but `session` is not nil and one of the workQueue items gets processed.

Since `session` is a property of `Adapter` I don't think this can happen.